pkgs/tools/package-management/nix-upfetch/default.nix

```diff
@@ -0,0 +1,102 @@
+{ stdenv, fetchFromGitHub, makeWrapper
+, rustPlatform, pkgconfig, ncurses
+, asciidoc, docbook_xml_dtd_45, docbook_xsl, libxml2, libxslt
+, libredirect, coreutils, gawk, gnugrep, gnused, jq, nix, nix-prefetch }:
+
+with stdenv.lib;
+
+let
+  nix-update-fetch = rustPlatform.buildRustPackage rec {
+    name = "${pname}-${version}";
+    pname = "nix-update-fetch";
+    version = "0.1.0";
+
+    src = fetchFromGitHub {
+      owner = "msteen";
+      repo = "nix-update-fetch";
+      rev = "bc201ec28be669f67501058e97b93e6a6e7b4e07";
+      sha256 = "09qqk9wcbbnwv0rkmr18k4pvaf6xpbrfci7ki44v8m6p4pwvqczw";
+    };
+
+    RUSTC_BOOTSTRAP = 1;
+
+    buildInputs = [ pkgconfig ncurses ];
+
+    cargoSha256 = "0g2gmmhx2gcb02yqmzavx7fqyvdblgg16rhq10rw2slnrmsz84k6";
+
+    meta = {
+      description = "Update a call made to a fetcher function call and its surrounding bindings";
+      homepage = https://github.com/msteen/nix-update-fetch;
+      license = licenses.mit;
+      maintainers = with maintainers; [ msteen ];
+      platforms = platforms.all;
+    };
+  };
+
+in stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "nix-upfetch";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "msteen";
+    repo = "nix-upfetch";
+    rev = "dd133c6a345710a285d5ed90c573b5b974e65c96";
+    sha256 = "1kmsr2j2qybrb8ksh5vq2zmx54gbsc56vcd6gs7jhswfmzhw0blc";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    asciidoc docbook_xml_dtd_45 docbook_xsl libxml2 libxslt
+  ];
+
+  configurePhase = ''
+    . configure.sh
+  '';
+
+  buildPhase = ''
+    a2x -f manpage doc/nix-upfetch.1.asciidoc
+  '';
+
+  installPhase = ''
+    lib=$out/lib/${pname}
```

Comment:
Why not create a makefile in the project?

Reply:
I have limited Makefile experience, what would this gain me? If it's worth doing, could you maybe give an example of what you have in mind?

Reply:
I think one reason might be that most of that code in `installPhase` could remain upstream. I get that it seems silly to you as you are upstream... but keeping code like that upstream wherever possible is the NixOS way.

We're so close on this PR. It would be nice to see it to completion.

Reply:
@aanderse If you want me to add a Makefile, I will, but I will not be attempting this myself without being pointed to some Makefile that does something similar to what I need to do, because I have only limited experience with making Makefiles (small C projects).

Reply:
@msteen Honestly I was just trying to triage old/stale PRs... @jtojnar Anything to add?

Reply:
If @msteen wants to make nix-upfetch available in different package repositories in the future, he can upstream his script and update this derivation.